cpp/src/arrow/compute/exec/key_hash.cc

```diff
@@ -198,8 +198,10 @@ void Hashing::hash_varlen_helper(uint32_t length, const uint8_t* key, uint32_t*
       last_stripe[1] = last_stripe_base[1];
       last_stripe[1] &= mask;
     }
+    uint32_t lanes[4];
+    memcpy(&lanes, &last_stripe, sizeof(last_stripe));
```

Comment:
Maybe add a comment as to why we're doing this (and mention the JIRA issue id)?

Reply:
Added the comment